SparkPost/2web2ui
# FE-1149 - [Domains UX Redesign] List page - Filtering UI

[FE-1149](https://sparkpost.atlassian.net/browse/FE-1149)

### What Changed
- Adds filtering UI for use in FE-1154
- Adds new `<LabelSpacer />` component to help with spacing problems in rows of fields
- Stylelint now runs against `.js` files only - the new rules are incompatible with .css files, and prior to a recent change, were not being run against them anyways so this results in no regression
- Resets `overflow` style on the content of each page - Cypress caught a bug in which a popover could flow outside of the bounds of that element, causing it to render as hidden
- Removed use of `useTabs` on the `/domains/list` page that was causing an infinite loop
- Adds some additional tests to check for a bug found in production while I'm here See: https://sparkpost.atlassian.net/browse/FE-1159

### How To Test
1. Sign in to an account with the account option `allow_domains_v2`
1. Go to `/domains` and verify the filtering UI renders on each tab
1. Verify filtering UI matches mocks (note, the `<Select/>` to filter by date should not be present - we will use regular table sorting)

### To Do
